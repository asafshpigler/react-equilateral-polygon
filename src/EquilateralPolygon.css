.equilateral-polygon-svg {
  overflow: visible;
}

.equilateral-polygon-polygon {
  fill: none;
  stroke: limegreen;
  stroke-width: 5px;
}