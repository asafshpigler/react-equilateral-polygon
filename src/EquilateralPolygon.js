import React, { Component } from 'react'
import {
  getPointsOfEquilateralPolygon,
  pointsArrayToString,
} from './utils.js'
import './EquilateralPolygon.css'

class EquilateralPolygon extends Component {
  constructor(props) {
    super()

    const { sideCount, size } = props
    const pointsStr = this.getPointsStr(sideCount, size)

    this.state = {
      points: pointsStr,
    }
  }

  componentDidUpdate(prevProps) {
    if (prevProps.sideCount === this.props.sideCount) return

    const { sideCount, size } = this.props
    const pointsStr = this.getPointsStr(sideCount, size)
    this.setState({ points: pointsStr })
  }

  getPointsStr(sideCount, size) {
    const isValidArgs  = typeof sideCount === 'number' && sideCount >= 3
    if (!isValidArgs) return

    const radius = size / 2
    const points = getPointsOfEquilateralPolygon(sideCount, radius)
    const pointsStr = pointsArrayToString(points)

    return pointsStr
  }

  render() {
    const { size } = this.props
    const { points } = this.state

    return points && (
      <svg
        xmlns="http://www.w3.org/2000/svg"
        className="equilateral-polygon-svg"
        style={{ height: size, width: size }}
      >
        <polygon
          className="equilateral-polygon-polygon"
          points={points}
        />
      </svg>
    )
  }
}

export default EquilateralPolygon