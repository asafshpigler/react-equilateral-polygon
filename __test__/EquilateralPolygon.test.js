// Link.react.test.js
import React from 'react'
import EquilateralPolygon from '../src/EquilateralPolygon.js'
import renderer from 'react-test-renderer'

test('EquilateralPolygon component is rendered', () => {
  const component = renderer.create(
    <EquilateralPolygon sideCount={3} size={50} />,
  )
  const tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})
