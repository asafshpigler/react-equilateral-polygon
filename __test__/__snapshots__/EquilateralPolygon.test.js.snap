// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EquilateralPolygon component is rendered 1`] = `
<svg
  class="equilateral-polygon-svg"
  style={
    Object {
      "height": 50,
      "width": 50,
    }
  }
  xmlns="http://www.w3.org/2000/svg"
>
  <polygon
    class="equilateral-polygon-polygon"
    points="25.00,0.00 46.65,37.50 3.35,37.50"
  />
</svg>
`;
